debug/org.eclipse.debug.tests/src/org/eclipse/debug/tests/launching/LaunchTests.java

```diff
@@ -146,6 +147,7 @@ public void testProcessLaunchWithLongWorkingDirectory() throws CoreException, IO
 		String[] segments = Collections.nCopies((400 - rootLength) / subPathElementsName.length(), subPathElementsName).toArray(String[]::new);
 		File workingDirectory = tempFolder.newFolder(segments);
 		assertTrue(workingDirectory.toString().length() > 300);
+		Files.createDirectories(workingDirectory.toPath());
 
 		startProcessAndAssertOutputContains(List.of("java", "--version"), workingDirectory, false, "jdk");
```

Comment:
Please note that "java --version" is wrong a as eclipse also supports j9. See org.eclipse.jdt.internal.launching.StandardVMType.CANDIDATE_JAVA_FILES. org.eclipse.jdt.internal.launching.LibraryInfo.getVersion() can be used to get the version

Reply:
The idea of this is to just launch any sub-process in a working directory with a path longer than Window's MAX_PATH length limit (~260 chars).
I choose `java --version` since it's a quite short running process and I assume java will be available by default on all test machines. But thinking more about it, that's probably the actual cause of the failure, not the working directory not being existent.